server/src/com/thoughtworks/go/config/update/RoleConfigCommand.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    private void validate(CruiseConfig preprocessedConfig) {
+        if (role instanceof PluginRoleConfig) {
+            PluginRoleConfig role = (PluginRoleConfig) this.role;
+            PluginRoleConfig preprocessedRole = (PluginRoleConfig) this.preprocessedRole;
+
+            SecurityAuthConfig securityAuthConfig = preprocessedConfig.server().security().securityAuthConfigs().find(role.getAuthConfigId());
+            if (securityAuthConfig == null) {
+                role.addError("authConfigId", "No such security auth configuration present " + role.getAuthConfigId());
+                return;
+            }
+            try {
+                ValidationResult result = extension.validateRoleConfiguration(securityAuthConfig.getPluginId(), role.getConfigurationAsMap(true));
+                if (!result.isSuccessful()) {
+                    for (ValidationError validationError : result.getErrors()) {
+                        ConfigurationProperty property = preprocessedRole.getProperty(validationError.getKey());
+                        if (property == null) {
+                            role.addNewConfiguration(validationError.getKey(), false);
+                            preprocessedRole.addNewConfiguration(validationError.getKey(), false);
+                            property = preprocessedRole.getProperty(validationError.getKey());
+                        }
+                        property.addError(validationError.getKey(), validationError.getMessage());
+
+                    }
+                }
+            } catch (PluginNotFoundException e) {
+                role.addError("authConfigId", "Could not find a security authorization config with id '" + role.getAuthConfigId() + "'.");
+            }
+        }
+    }
+
+
+    final Role findExistingRole(CruiseConfig cruiseConfig) {
+        if (role == null || isBlank(role.getName().toString())) {
+            if (role != null) {
+                role.addError("name", "Plugin role config cannot have a blank name.");
+            }
+            result.unprocessableEntity(LocalizedMessage.string("ENTITY_ATTRIBUTE_NULL", "plugin role config", "name"));
+            throw new IllegalArgumentException("Plugin role config name cannot be null.");
+        } else {
+            Role t = cruiseConfig.server().security().getRoles().findByName(role.getName());
+            if (t == null) {
+                result.notFound(LocalizedMessage.string("RESOURCE_NOT_FOUND", getTagName(), role.getName()), HealthStateType.notFound());
+                throw new RoleNotFoundException("Plugin role config `" + role.getName() + "` does not exist.");
+            }
+            return t;
+        }
+    }
+
+    private String getTagName() {
+        return role.getClass().getAnnotation(ConfigTag.class).value();
+    }
+
+    protected final boolean isAuthorized() {
+        if (!(goConfigService.isUserAdmin(currentUser) || goConfigService.isGroupAdministrator(currentUser.getUsername()))) {
```

Comment:
Only super-admins should be allowed this operations. Group admins is purely for managing pipeline groups.